src/audio/buffer.c

```diff
@@ -27,6 +27,7 @@ uintptr_t buffer_uuid_ptr = SOF_UUID(buffer_uuid);
 struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align)
 {
 	struct comp_buffer *buffer;
+	uint32_t mem_caps = caps & ~0xFFFF0000;
```

Comment:
We need a macro for this mask value. It's also worth an inline comment here too.

Reply:
```suggestion
	uint32_t mem_caps = caps & 0xFFFFL;
```
This looks more straight forward for only need lower 16bits

Reply:
added now 0xFFFF in a macro, seems to work.